controllers/ReviewController.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file Controller RESTful Web service API for reviews resource
+ */
+import {Request, Response, Express} from "express";
+import ReviewDao from "../daos/ReviewDao";
+import ReviewControllerI from "../interfaces/ReviewControllerI";
+
+/**
+ * @class TuitController Implements RESTful Web service API for reviews resource
+ * Defines the following HTTP endpoints:
+ * <ul>
+ *     <li>GET /api/reviews to retrieve all the reviews instances</li>
+ *     <li>GET /api/restaurants/:restaurantid/reviews to retrieve reviews for a given restaurant </li>
+ *     <li>GET /api/users/:criticid/reviews to retrieve reviews by a given critic </li>
+ *     <li>GET /api/reviews/:reviewid to retrieve a particular review instance</li>
+ *     <li>POST /api/restaurants/:restaurantid/users/:criticid/reviews to create a new review instance</li>
+ *     <li>PUT /api/reviews/:reviewid to modify an individual review instance </li>
+ *     <li>DELETE /api/reviews/:reviewid to remove a particular review instance</li>
+ * </ul>
+ * @property {ReviewDao} reviewDao Singleton DAO implementing review CRUD operations
+ * @property {ReviewController} review Controller Singleton controller implementing RESTful Web service API
+ */
+export default class ReviewController implements ReviewControllerI {
+    private static reviewController: ReviewController | null = null;
+    private static reviewDao: ReviewDao = ReviewDao.getInstance();
+
+    /**
+     * Creates singleton controller instance
+     * @param {Express} app Express instance to declare the RESTful Web service API
+     * @return ReviewController
+     */
+    public static getInstance = (app: Express): ReviewController => {
+        if (ReviewController.reviewController === null) {
+            ReviewController.reviewController = new ReviewController();
+            app.get("/api/reviews", ReviewController.reviewController.findAllReviews);
+            app.get("/api/restaurants/:restaurantid/reviews", ReviewController.reviewController.findAllReviewsForRestaurant);
+            app.get("/api/users/:criticid/reviews", ReviewController.reviewController.findAllReviewsByCritic);
+            app.get("/api/reviews/:reviewid", ReviewController.reviewController.findReviewById);
+            app.post("/api/restaurants/:restaurantid/users/:criticid/reviews", ReviewController.reviewController.createReview);
+            app.put("/api/reviews/:reviewid", ReviewController.reviewController.updateReview);
+            app.delete("/api/reviews/:reviewid", ReviewController.reviewController.deleteReview);
+        }
+        return ReviewController.reviewController;
+    }
+    private constructor() {}
+
+    /**
+     * Creates a new review instance
+     * @param {Request} req Represents request from client, including path
+     * parameter criticid identifying the primary key of the user that created
+     * the review, reviewid identifyng the primary key of the restaurant being reviewed
```

Comment:
Parameter name is actually restaurantid and not reviewid - do you mind updating this? And there's also a small typo in "identifying" right after reviewid